analysers/analyser_osmosis_indoor.py

```diff
@@ -92,6 +125,48 @@
 # and assuming than no level on highway is probably implicit level=0)
 
 sql21 = """
+CREATE TEMP TABLE indoor_pt_platforms AS
+SELECT
+    relations.id,
+    ST_MakePolygon(outer_ways.linestring,array_agg(inner_ways.linestring)) AS geom
+FROM
+    relations AS relations
+    JOIN relation_members AS outer_members ON
+        outer_members.relation_id = relations.id AND
+        outer_members.member_type = 'W' AND
+        outer_members.member_role = 'outer'
+    JOIN ways AS outer_ways ON
+        outer_ways.id = outer_members.member_id AND
+        ST_NumPoints(outer_ways.linestring) >= 2
+    JOIN relation_members AS inner_members ON
+        inner_members.relation_id = relations.id AND
+        inner_members.member_type = 'W' AND
+        inner_members.member_role = 'inner'
+    JOIN ways AS inner_ways ON
+        inner_ways.id = inner_members.member_id AND
+        ST_NumPoints(inner_ways.linestring) >= 2
+WHERE
+    relations.tags?'type' AND
+    relations.tags->'type' = 'multipolygon' AND
+    relations.tags?'public_transport' AND
+    relations.tags->'public_transport' = 'platform'
+GROUP BY
+    relations.id,
+    outer_ways.id
+UNION ALL
+SELECT
+    ways.id,
+    ways.linestring AS geom
+FROM
+    ways
+WHERE
+    ways.is_polygon AND
```

Comment:
Even when is_polygon, linestring is a ...linestring. You have to use `ST_MakePolygon` on it.